src/filter/filter_gain.rs

```diff
@@ -0,0 +1,84 @@
+use nalgebra as na;
+use super::super::config::*;
+
+// pub fn measurement_vector(
+//     jacobian: &Mat5,
+//     previous_state_vec: &Vec5) -> Vec2{
+//     return jacobian * previous_state_vec
+// }
+
+pub fn state_vector( // x
+    extrap_state_vector: &Vec5, 
+    kalman_gain: &Mat5x2, 
+    measurement : &Vec2, 
+    sensor_mapping: &Mat2x5) -> Vec5 {
+
+    let parens = measurement - (sensor_mapping * extrap_state_vector);
+    let kalman_product = kalman_gain * parens;
+    return extrap_state_vector + kalman_product;
+}
+
+//TODO: remove `unwrap` on the inverse
+#[allow(dead_code)]
+pub fn kalman_gain ( // K
+    C : &Mat5,
+    H : &Mat2x5,
+    V : &Mat2) -> Mat5x2 {
+
+    let parens = V + ( H * C * H.transpose() );
+    let kalman_gain = C * H.transpose() * parens.try_inverse().unwrap();
+    
+    kalman_gain
+}
+
+//TODO add lazy static for identity
+//update covariance matrix C
+pub fn covariance_matrix( // C
+    K : &Mat5x2,
+    H : &Mat2x5,
+    C : &Mat5) -> Mat5 {
+    let parens = Mat5::identity() - (K*H);
+
+    return C * parens;
+}
+
+//TODO  add second method for this calculation
+//      since the second one looks to have less matmul
+
+//TODO add lazy static for identity
+pub fn residual_vec(  //r
+    H : &Mat2x5,
+    K : &Mat5x2,
+    residual_preiction : &Vec2) -> Vec2 {
+
+    let ident = Mat2::identity();
+    let parens = ident - (H * K);
+
+    return  parens * residual_preiction;
+}
+
+pub fn residual_mat( //R
```

Comment:
I just tried to keep the naming somewhat consistent between `prediction.rs` `filter_gain.rs` `filter_mean.rs` and `smoothing.rs` and parallel to the paper's description. 
I thought it might make sense to see not name it `filtered_residual_matrix` since it is already in `filter_gain.rs`. How would you name it?

Reply:
I see what you mean. I don't have an immediate suggestion on what to call it instead, so let's keep it for now.